common/autoware_auto_perception_rviz_plugin/include/object_detection/predicted_objects_display.hpp

```diff
@@ -96,24 +98,64 @@ class AUTOWARE_AUTO_PERCEPTION_RVIZ_PLUGIN_PUBLIC PredictedObjectsDisplay
     return id_map.at(uuid);
   }
 
+  // createMarkers: create markers from the message with the single working process; activated when
+  // num_threads<=1;
   std::vector<visualization_msgs::msg::Marker::SharedPtr> createMarkers(
     PredictedObjects::ConstSharedPtr msg);
+  // Working thread is detached from the main to control the flow of messages
   void workerThread();
+  // parallelizedCreateMarkerWorkerThread: create markers from the message with multiple workers
+  void parallelizedCreateMarkerWorkerThread(int rank);
 
   void update(float wall_dt, float ros_dt) override;
 
+  std::vector<visualization_msgs::msg::Marker::SharedPtr> tackle_one_object(
+    PredictedObjects::ConstSharedPtr _msg, int index,
+    std::vector<visualization_msgs::msg::Marker::SharedPtr> _markers);
```

Comment:
```suggestion
    PredictedObjects::ConstSharedPtr msg, int index,
    std::vector<visualization_msgs::msg::Marker::SharedPtr> markers);
```

Minor suggestion (religious suggestion): If you always set underbar to the local variable, `_msg` and `_markers` look fine, but this time the rule is not in order.

Reply:
This will cause quite a lot of changes in the cpp file. If that is better I will push another commit here.

Reply:
yeah...then this time, let's ignore this